sourcemap/fleet_test.go

```diff
@@ -123,9 +152,21 @@ func TestFailedAndSuccessfulFleetHostsFetch(t *testing.T) {
 	f, err := NewFleetFetcher(c, fleetCfg, cfgs)
 	assert.NoError(t, err)
 
-	consumer, err := f.Fetch(context.Background(), name, version, path)
-	assert.NoError(t, err)
-	assert.NotNil(t, consumer)
+	go func() {
+		consumer, err := f.Fetch(context.Background(), name, version, path)
+		resc <- response{consumer, err}
+	}()
+	// Make sure every server has received a request
+	wg.Wait()
```

Comment:
Can't we remove this wait group? It seems to me that the channels are already making the test wait until we've received all the signals. What is this wait group adding on top of that?

Reply:
The waitgroup is making sure that each handler has received a request before we attempt to read from any of the channels. Without that, we could potentially read from `errc` and `successc` without the third handler ever being invoked.

Reply:
got it, feel free to merge, thanks for the explanation